### DISCUSSION TITLE
help request: 自定义插件复制jwt-auth实现出现 Missing related consumer

### DISCUSSION THREAD

**Discussion by wulinghui** [apache/apisix#8075] [Q&A]
### Description

我按钮文档去新增了一个lua插件并且配置使得该插件生效。
但是在访问对应的路由时他报 `{"message":"Missing related consumer"}`

我看源码代码是
```
    local consumer_conf = consumer_mod.plugin(plugin_name)
    if not consumer_conf then
        return 401, {message = "Missing related consumer"}
    end
```

是不是consumer还需要再把目前的插件名给配置上去？？

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by tokers** [comment]
The information you provided was totally useless. You should tell us which kinds of plugin are you trying to create. Also, if possible, paste some code snippets here.

Looks like you're writing an auth plugin, an auth plugin should design a way to find an APISIX consumer via the current API request (e.g., via a header, an arg, or even from the request body). In addition, the plugin type should be `auth`.

**Comment by wulinghui** [comment]
[jwt-pluns.zip](https://github.com/apache/apisix/files/9753745/jwt-pluns.zip)
ok,谢谢回复。 请看压缩包里面。 
我添加了一个插件（代码完全复制jwt-auth.lua）只是修改了name，并在config-default.yaml 配置文件中了启用的插件该插件。
也的确生效了，但是输入了`?jwt=xxxxxxxxx`后就报 `{message = "Missing related consumer"}` 错误啦。

**Reply by tzssangglass** [reply]
Do you have a consumer configured and does this consumer use the plugin you copied?

**Reply by wulinghui** [reply]
ok，Thank you. It will be fine if it is configured by the consumer
